### ISSUE TITLE
Add "Configure page title" tab to Layout admin pages

### ISSUE THREAD

**Issue by jackaponte** [backdrop/backdrop-issues#3983]
**Description of the need**
To configure a layout's page title, you need to visit the "Configure page title" page at `/admin/structure/layouts/manage/<layout-machine-name>/edit-title/editor/title`. A menu item for that admin page appears in the Backdrop admin menu, but that's easy to miss; no tab appears on any of the other Layout admin pages to let you reach the "Configure page title" page.

**Proposed solution**
Like "Manage blocks" and "Configure layout," a "Configure page title" tab should appear at the top of each layout's admin pages.

**Comment by jackaponte** [comment]
I realize now that when on the Manage Blocks screen there's a "Page title" element that you can configure, but since there's a menu item for "Configure page title" and there are tabs for all the other menu items, seems like consistency is best! (Also, that "Page title" element was easy for me to miss!)

**Comment by klonos** [comment]
> To configure a layout's page title, you need to visit the "Configure page title" page at `/admin/structure/layouts/manage/<layout-machine-name>/edit-title/editor/title`. ... I realize now that when on the Manage Blocks screen there's a "Page title" element that you can configure ...

Yeah, you get to the same page if you try to click the "Configure" button of that "Page title" pseudo-block in the "Manage blocks" page; only that opens in a modal dialog. If you have js disabled on your browser, or if you right-click the "Configure" button and select to open in a different browser tab, you'll get to the `/admin/structure/layouts/manage/<layout-machine-name>/edit-title/editor/title` page 😉 

> ...that "Page title" element was easy for me to miss!

Yeah, I guess this is one of those "if you know it, you know it" things 😅 ...having said that though, one could then also request a separate menu and a tab for the breadcrumb block, and for any other block. It would get really messy really soon. The "Page title" thingy is a special-case, pseudo block, and that's why it has a menu item in the Admin Bar (🍻 🍸 🍹 😄 ); I don't think that we should have a separate tab for it.

**Comment by jackaponte** [comment]
> one could then also request a separate menu and a tab for the breadcrumb block, and for any other block. It would get really messy really soon. The "Page title" thingy is a special-case, pseudo block, and that's why it has a menu item in the Admin Bar (beers cocktail tropical_drink smile ); I don't think that we should have a separate tab for it.

Because Page Title is a special case, pseudo block with a menu item is precisely why I think it should also get a tab up top, whereas other blocks on the page should not! Like, if it has a menu item, it should have a tab too, for consistency with the other menu items that have tabs. Also because Page Title is one of the most important elements of any page being created, I think that merits making it as visible and accessible as possible.

**Comment by klonos** [comment]
Fair enough; I am not vigorously opposed to this. Curious to know what others think.

**Comment by bugfolder** [comment]
I just ran into this and was about to post an issue, then found this one. I very much agree with @jackaponte; it should get a tab for consistency and visibility/accessibility.

**Comment by bugfolder** [comment]
I"d like to move this along, so I've filed a PR. @klonos, does this still need the "needs more feedback" tag?

**Comment by bugfolder** [comment]
> I"d like to move this along, so I've filed a PR. @klonos, does this still need the "needs more feedback" tag?

Since a few months have gone by with no further feedback, I'm going to remove the "needs more feedback" tag and reiterate the request for testing and code review.

**Comment by klonos** [comment]
@bugfolder sorry for the late reply here. I think that the "needs feedback" label is valid, since there's only 3 people that have chimed in on this thread.

The code change and implementation is very simple, but the question is if this is a good change/implementation. Can you please close/reopen the PR (and possibly also rebase) to get the sandbox going? Thanks.

**Comment by bugfolder** [comment]
OK, closed and reopened PR, restore NMF label. (Perhaps this will provide a jog for folks to get more feedback!)

**Comment by bugfolder** [comment]
And rebased, because PHPCS.

**Comment by klonos** [comment]
Thanks @bugfolder 🙏🏼 ...code looks simple and good, and it works as advertized 👍🏼 

The only thing is that we now have a couple of ways to configure the same thing:
![image](https://user-images.githubusercontent.com/2423362/210187533-d9e70485-71ae-44d9-a965-b668ff746a6d.png)
...should we remove the "Page title" pseudo block in favor of the new tab, or keep both? 🤔

**Comment by bugfolder** [comment]
Since they both say "configure", I see no harm in having two ways to get there. Actually, there's three ways, since "Configure page title" also appears in the admin menu (which was part of the original impetus for this issue). But let's hear other folks.

**Comment by quicksketch** [comment]
I'm not sure about this change. We're adding the same option, on the same form, both above the fold (though that depends on how many blocks you have above the page title). There's also a fairly good chance of conflicting settings between the Manage Blocks form and the Configure Page Title form. Take the following scenario:

- Start modifying blocks so the form changes to in-progress.
- Visit Configure Page Title and modify the setting and save.
- Visit manage blocks again, continue modifying blocks.
- Save.

Now what may happen is that the in-progress setting for Page Title will revert the page title setting that was changed on the other form.

However, in testing the current PR, I found this doesn't happen (yet) because the Configure Page Title form still has a `#ajax` handler on it, so saving the form doesn't do anything at all until you go to the Manage Blocks form and complete the save.

**Comment by argiepiano** [comment]
I'm not opposed to adding the tab - even if you end up with two ways of configuring the page title. However, there are some important caveats for me, related to the way this title configuration works (perhaps needing a separate issue): 

If your layout has the pseudo-block "Page title" and you click "Configure", you will be shown a modal that allows you to configure the title 👍🏽 . 

On the other hand, if you have a "Page title combo" block, the pseudo-block disappears, and you click "Configure", you won't see the title configuration modal anymore. 👎🏽  The title configuration is now found in the dropdown under "Page title settings" - I find this confusing. In a layout where I switched from the pseudo-block to the title combo, I thought for a while that I wouldn't be able to configure the title anymore (I kept clicking "Configure" and not seeing the title config options), until I found the other menu item in the dropdown.

Finally (somewhat unrelated to this issue): if you access the title configuration form by hovering over it in the admin menu (Structure > Layout > [your layout], you are taken to a form page to configure the title, not a modal anymore. Submitting that form doesn't redirect you to the "Manage block" page of the layout, even though you are supposed to save the layout for any title changes to take effect. This is really confusing, and will be the behavior of this page once this is added as a tab as well. 

So if we end up adding the tab, I'd suggest that submitting the form page should redirect you to the Manage block page for that layout, so that you are told that changes need to be saved to take effect.

**Comment by olafgrabienski** [comment]
As @quicksketch, I'm also not sure if we should add the tab. First, to me it's not clear why "Configure page title" with its narrow scope appears in the admin menu at all. Not that I want to remove it, but I don't think it's *that* important; I see the block configuration page as standard place to configure the page title.

Second, the addition of the tab is based on the following assumption:

> since there's a menu item for "Configure page title" and there are tabs for all the other menu items, seems like consistency is best

In my opinion, the assumption isn't sufficient, as not every admin menu entry in Backdrop gets a tab.

**Comment by klonos** [comment]
I think that the actual UX problem here is that the title configuration and the title pseudo-block are not easy to spot in the block management UI, and also that the pseudo-block makes sense from a dev point of view (once you know its purpose and the "mechanics" behind it), otherwise, this non-draggable thing among other draggable things is confusing.

Is there any other way to improve that situation? Can we (and should we) make the title pseudo block more prominent? Should we try to merge/consolidate the functionality of the pseudo-block with the Title and Title combo blocks? (too many similar things too choose from makes it confusing) 🤔

**Comment by kiamlaluno** [comment]
I would leave the pseudo-block to move the title in the layout, remove its *Configure* button, and use the new tab to configure it.

**Comment by argiepiano** [comment]
> I would leave the pseudo-block to move the title in the layout

The title pseudo-block can't be moved in the layout.

**Comment by kiamlaluno** [comment]
> The title pseudo-block can't be moved in the layout.

That's correct. It's more correct to say *Use the pseudo-block to only show where the title will be rendered in the page.*

I meant that the pseudo-block has its purpose, even if it doesn't have a *Configure* button.

**Comment by klonos** [comment]
Right. I will need to consider the UX a bit more on this 🤔

**Comment by yorkshire-pudding** [comment]
I was so confused by this. The form behaviour when you access from the admin menu is really confusing.
![chrome_MhFHJ9EQk0](https://github.com/backdrop/backdrop-issues/assets/20357797/da920abf-27fa-43fb-b560-7242d1eac128)
It feels like the form is broken - see also #4707 . If a tab would suffer the same issue then it wouldn't be good. In the pseudo block when it is Page title combo then Configure deals with styling and local tabs, and has a separate menu link (in the drop button) for "page title settings" that brings up the ajax dialog (which works as expected).

I don't think it should be in the admin menu as it is but we should perhaps make it part of the configure menu

> Is there any other way to improve that situation? Can we (and should we) make the title pseudo block more prominent? Should we try to merge/consolidate the functionality of the pseudo-block with the Title and Title combo blocks? (too many similar things to choose from makes it confusing)

I like this idea from @klonos - Have a single title block with configuration in a dialog for everything but only revealing settings when options are enabled.

**Comment by klonos** [comment]
Assigning this to myself, but since I'm working on a few other issues I'm trying to get in, please feel free to grab it if anyone has capacity to work on this before I get to it.

**Comment by klonos** [comment]
...oh, sorry, I just noticed now that there's already a PR by @bugfolder. Assigning to you then 😉

**Comment by bugfolder** [comment]
Oh, feel free to take it over, @klonos.

**Comment by klonos** [comment]
> Right. I will need to consider the UX a bit more on this 🤔

Yeah. This may seem like an easy thing, but it is far from it. Depending on your expectations of how the UI should work + the task you set out to perform + your perception of the various UI patterns we are using for similar things + your knowledge of the layout "internals" (how the different variations of title blocks work and what their purpose is), then adding a new horizontal tab and/or keeping or removing the menu items from the admin menu may either make things better or worse for some people (introduce more complexity/confusion).

It really feels like we are exposing unnecessary complexity to possibly less technical people. Sure, the people coming to the layout configuration forms must have some rather advanced knowledge (being site builders) if they are coming here to tweak things, but I don't want this UI to end up having the same complexity reputation as the Views UI (don't have to be a developer to know how to configure things).

So I would really like us to consider moving everything into a single modal dialog, in a "one form to rule them all" approach for everything related to layout titles. We could then have the modal be showing only the options relevant to how things have been configured + help text or info messages that explain things if needed.

Still need to think this through 🤔 ...tagging as a `design` topic.

**Comment by bugfolder** [comment]
FWIW, I resolved the conflicts in the existing PR just to leave things neat and tidy, but still happy to hand this over to @klonos.
